ballerina/json_schema.bal

```diff
@@ -0,0 +1,125 @@
+    # The number must be a multiple of this value.
+    int|float|decimal multipleOf?;
+|};
+
+# The annotation is used to specify validation constraints for number type.
+public annotation NumberConstraintsConfig NumberConstraints on type;
+
+# Defines constraints for JSON Schema array type.
+public type ArrayConstraintsConfig record {|
+    # The minimum number of items required in the array.
+    int:Unsigned32 minItems = 0;
+    # The maximum number of items allowed in the array.
+    int:Unsigned32 maxItems?;
+    # Specifies whether all items in the array must be unique.
+    boolean uniqueItems?;
+    # Specifies a schema that a specific number of items in the array must match.
+    record {|
+        int:Unsigned32 minContains = 0;
+        int:Unsigned32 maxContains?;
+        typedesc<json> value;
+    |} contains?;
+    # Specifies a schema for items that are not evaluated by contains.
+    typedesc<json> unevaluatedItems?;
+|};
+
+# The annotation is used to specify validation constraints for array type.
+public annotation ArrayConstraintsConfig ArrayConstraints on type;
+
+# Defines constraints for JSON Schema object type.
+public type ObjectConstraintsConfig record {|
+    # The maximum number of properties allowed in the object.
+    int:Unsigned32 maxProperties?;
+    # The minimum number of properties required in the object.
+    int:Unsigned32 minProperties?;
+    # The schema to validate the names of properties in the object.
+    typedesc<json> propertyNames?;
+|};
+
+# The annotation is used to specify validation constraints for object type.
+public annotation ObjectConstraintsConfig ObjectConstraints on type;
+
+# Defines a dependent sub Schema in an object field.
+public type DependentSchemaConfig record {|
+    # The dependent sub Schema
+    typedesc<json> value;
+|};
+
+# The annotation is used to specify a dependent schema.
+public annotation DependentSchemaConfig DependentSchema on record field;
+
+# Defines the fields dependent required on other fields.
+public type DepenedentRequiredConfig record {|
+    # The dependent required field/fields
+    string|string[] value;
+|};
+
+# The annotation is used to specify dependent Required fields.
+public annotation DepenedentRequiredConfig DependentRequired on record field;
+
+# Represents a single patternProperties rule in a JSON object schema.
+public type PatternPropertiesElement record {|
+    # A regular expression that the property name must match
+    string:RegExp pattern;
+    # The type that properties matching the pattern must conform to
+    typedesc<json> value;
+|};
+
+# A container for multiple patternProperties rules.
+public type PatternPropertiesElements record {|
+    # The list of patternProperties rules (each with a pattern and type)
+    PatternPropertiesElement|PatternPropertiesElement[] value;
+|};
+
+# Annotation used to define patternProperties constraints on object types.
+public annotation PatternPropertiesElements PatternProperties on type;
```

Comment:
Same comment as my previous comment re: using terms like patternProperties in user docs.